### PR TITLE
Use our server-side Google Classroom endpoint instead of the JS API

### PR DESCRIPTION
**Pull request by joshlory**
This change builds on https://github.com/code-dot-org/code-dot-org/pull/16253 to load Google Classroom data via the Ruby API instead of the JavaScript API.  The excluding that PR, the relevant diff is here: https://github.com/code-dot-org/code-dot-org/compare/oauth-save-refresh-token...google-classroom-server-side.

API errors are reported inside the `RosterDialog` component:

screen shot 2017-07-07 at 4 32 27 pm

~TODO:~
 - [x] Merge staging into this branch after PR https://github.com/code-dot-org/code-dot-org/pull/16253 is merged.
 - [x] Pull `studioUrl` out of the Redux store.

### CODE REVIEW

File: apps/src/templates/teacherDashboard/RosterDialog.jsx [hunk 1]
```diff
@@ -88,11 +88,21 @@ ClassroomList.propTypes = {
   selectedId: React.PropTypes.string,
 };
 
+const LoadError = () =>
+  <div>
+    <a href="//localhost-studio.code.org:3000/users/auth/google_oauth2?scope=classroom.courses.readonly,classroom.rosters.readonly">
```
Comment (joshlory) [inline apps/src/templates/teacherDashboard/RosterDialog.jsx]: TODO: Pull studioUrl out of the Redux store.

File: apps/src/templates/teacherDashboard/RosterDialog.jsx [hunk 2]
```diff
@@ -88,11 +88,21 @@ ClassroomList.propTypes = {
...
+  </div>;
+
 export default class RosterDialog extends React.Component {
   static propTypes = {
     handleClose: React.PropTypes.func,
     isOpen: React.PropTypes.bool,
-    classrooms: React.PropTypes.arrayOf(classroomShape),
+    classrooms: React.PropTypes.oneOf(
+      React.PropTypes.arrayOf(classroomShape),
+      React.PropTypes.bool,
```
Comment (Bjvanminnen) [inline apps/src/templates/teacherDashboard/RosterDialog.jsx]: This feels like a weird type definition. Could we instead have classrooms be null or undefined?

**Comment by Bjvanminnen** [comment]
generally lgtm. might be worth having someone with more expertise around oauth/devise to review that stuff.

**Comment by joshlory** [comment]
Ok, I've merged in the other PR (the OAuth diff should disappear).  PTAL!

File: apps/src/templates/teacherDashboard/RosterDialog.jsx [hunk 3]
```diff
@@ -123,13 +143,18 @@ export default class RosterDialog extends React.Component {
...
+              studioUrl={this.props.studioUrl}
             /> :
-            locale.loading()
+            this.props.classrooms ?
+              <ClassroomList
+                classrooms={this.props.classrooms}
+                onSelect={this.onClassroomSelected}
+                selectedId={this.state.selectedId}
+              /> :
+              locale.loading()
```
Comment (Bjvanminnen) [inline apps/src/templates/teacherDashboard/RosterDialog.jsx:157]: I tend to prefer avoiding ternaries in JSX as I find it them difficult to read/follow (nested ternaries even more so).

**Pull request merged**